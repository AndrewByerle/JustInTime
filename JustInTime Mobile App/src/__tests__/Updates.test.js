import { render, screen } from '@testing-library/react-native';
import UpdatesScreen from './../../src/navigation/screens/Updates';

jest.useFakeTimers();

test('integration check that title shows on updates page', () => {
    render(<UpdatesScreen />);
    expect(screen.getByText('Recent Updates')).toBeTruthy();
});

test('check that date bar shows on updates page', () => {
    render(<UpdatesScreen />);
    expect(
        screen.getByText(`${new Date().getDate()}`, {
            exact: false,
        })
    ).toBeTruthy();
});
