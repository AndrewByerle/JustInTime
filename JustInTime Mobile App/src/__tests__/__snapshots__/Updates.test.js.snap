// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check for same updates page snapshot 1`] = `
<View>
  <View
    style={
      [
        {
          "paddingHorizontal": 24,
        },
        {
          "paddingTop": 56,
        },
        {
          "paddingBottom": 8,
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "#4B9CD3",
          "fontSize": 24,
        }
      }
    >
      Monday, November 28
    </Text>
  </View>
  <View
    style={
      {
        "paddingHorizontal": 24,
        "paddingVertical": 0,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 36,
          },
          {
            "fontWeight": "bold",
          },
          {
            "color": "#4B9CD3",
          },
        ]
      }
    >
      Recent Updates
    </Text>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <RCTScrollView
      alwaysBounceVertical={false}
      contentContainerStyle={
        {
          "paddingBottom": 30,
        }
      }
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={true}
        />
      }
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      style={
        {
          "marginBottom": 265,
          "minHeight": 100,
        }
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <RCTRefreshControl />
      <View />
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
